editoast/src/models/pathfinding.rs

```diff
@@ -0,0 +1,101 @@
+//! Provides the [Pathfinding] model
+
+use crate::schema::Direction;
+use crate::tables::osrd_infra_pathmodel;
+use chrono::{NaiveDateTime, Utc};
+use derivative::Derivative;
+use diesel::prelude::*;
+use diesel::result::Error as DieselError;
+use diesel::ExpressionMethods;
+use diesel::QueryDsl;
+use editoast_derive::Model;
+use postgis_diesel::types::*;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// Describes a pathfinding that resulted from a simulation and stored in the DB
+///
+/// It differs from infra/pathfinding that performs a topological pathfinding
+/// meant to be used with the infra editor.
+#[derive(
+    Debug, Clone, Serialize, Insertable, Derivative, Queryable, QueryableByName, AsChangeset, Model,
+)]
+#[derivative(Default(new = "true"))]
+#[model(table = "osrd_infra_pathmodel")]
+#[model(create, delete, retrieve, update)]
+#[diesel(table_name = osrd_infra_pathmodel)]
+pub struct Pathfinding {
+    pub id: i64,
+    pub owner: uuid::Uuid,
+    #[derivative(Default(value = "Utc::now().naive_utc()"))]
+    pub created: NaiveDateTime,
+    #[derivative(Default(value = "diesel_json::Json::new(Default::default())"))]
+    pub payload: diesel_json::Json<PathfindingCorePayload>,
+    #[derivative(Default(value = "diesel_json::Json::new(Default::default())"))]
+    pub slopes: diesel_json::Json<SlopeGraph>,
+    #[derivative(Default(value = "diesel_json::Json::new(Default::default())"))]
+    pub curves: diesel_json::Json<CurveGraph>,
+    #[derivative(Default(value = "LineString { points: vec![], srid: None }"))]
+    pub geographic: LineString<Point>,
+    #[derivative(Default(value = "LineString { points: vec![], srid: None }"))]
+    pub schematic: LineString<Point>,
+    pub infra_id: i64,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct SlopeGraph(Vec<Slope>);
```

Comment:
We don't need this struct. We can declare a `type` if you want this abstraction.

Reply:
Good point.